checks/evaluation/cii_best_practices.go

```diff
@@ -15,48 +15,93 @@
 package evaluation
 
 import (
-	"fmt"
-
 	"github.com/ossf/scorecard/v4/checker"
-	"github.com/ossf/scorecard/v4/clients"
 	sce "github.com/ossf/scorecard/v4/errors"
+	"github.com/ossf/scorecard/v4/finding"
+	"github.com/ossf/scorecard/v4/probes/hasGoldBadge"
+	"github.com/ossf/scorecard/v4/probes/hasInProgressBadge"
+	"github.com/ossf/scorecard/v4/probes/hasPassingBadge"
+	"github.com/ossf/scorecard/v4/probes/hasSilverBadge"
+	"github.com/ossf/scorecard/v4/probes/hasUnknownBadge"
 )
 
-// Note: exported for unit tests.
 const (
 	silverScore = 7
 	// Note: if this value is changed, please update the action's threshold score
 	// https://github.com/ossf/scorecard-action/blob/main/policies/template.yml#L61.
 	passingScore    = 5
 	inProgressScore = 2
+	maxScore        = 10
+	minScore        = 0
+	errScore        = -1
 )
 
 // CIIBestPractices applies the score policy for the CIIBestPractices check.
-func CIIBestPractices(name string, _ checker.DetailLogger, r *checker.CIIBestPracticesData) checker.CheckResult {
-	if r == nil {
-		e := sce.WithMessage(sce.ErrScorecardInternal, "empty raw data")
+func CIIBestPractices(name string,
+	findings []finding.Finding, dl checker.DetailLogger,
+) checker.CheckResult {
+	expectedProbes := []string{
+		hasGoldBadge.Probe,
+		hasSilverBadge.Probe,
+		hasInProgressBadge.Probe,
+		hasPassingBadge.Probe,
+		hasUnknownBadge.Probe,
+	}
+
+	if !finding.UniqueProbesEqual(findings, expectedProbes) {
+		e := sce.WithMessage(sce.ErrScorecardInternal, "invalid probe results")
+		return checker.CreateRuntimeErrorResult(name, e)
+	}
+
+	var countBadges int
+	// Sanity check that we don't have multiple positives
+	for i := range findings {
+		f := &findings[i]
+
+		if f.Outcome == finding.OutcomePositive {
+			countBadges++
+		}
+	}
+
+	var score int
+	var text string
+
+	if countBadges > 1 {
+		errText := "invalid probe results: multiple badges detected"
+		e := sce.WithMessage(sce.ErrScorecardInternal, errText)
 		return checker.CreateRuntimeErrorResult(name, e)
+	} else if countBadges == 0 {
+		text = "no effort to earn an OpenSSF best practices badge detected"
+		return checker.CreateResultWithScore(name, text, minScore)
```

Comment:
nit: don't use `checker.CreateResultWithScore` with a `minScore` constant. That's what `checker.CreateMinScoreResult` is for

Reply:
Done in https://github.com/ossf/scorecard/pull/3520/commits/1925fcaaa3b3309ae9b347b7790d122e8ed1305d